### PR TITLE
[WIP] Add exception class for fixer mis configuration.

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

File: Symfony/CS/Config/InvalidFixerConfigurationException.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Config;
+
+class InvalidFixerConfigurationException extends \Exception
```
Comment (SpacePossum) [inline Symfony/CS/Config/InvalidFixerConfigurationException.php]: maybe a better parent class would be `\UnexpectedValueException` or `InvalidArgumentException`, any thoughts?

Comment (keradus) [inline Symfony/CS/Config/InvalidFixerConfigurationException.php]: indeed

File: Symfony/CS/FixerInterface.php [hunk 2]
```diff
@@ -27,6 +28,8 @@
      * Finally, some fixers need configuration to work, eg `header_comment`.
      *
      * @param array|null $configuration configuration depends on Fixer
+     *
+     * @throws InvalidFixerConfigurationException
```
Comment (keradus) [inline Symfony/CS/FixerInterface.php]: we aren't using that annotation

File: Symfony/CS/Config/InvalidFixerConfigurationException.php (hunk 1 above)
Comment (keradus) [inline Symfony/CS/Config/InvalidFixerConfigurationException.php]: invalid one

**Comment by SpacePossum** [comment]
rebased and cleanupped

File: Symfony/CS/Fixer.php [hunk 3]
```diff
@@ -83,7 +83,10 @@ public function registerBuiltInConfigs()
         foreach (Finder::create()->files()->in(__DIR__.'/Config') as $file) {
             $relativeNamespace = $file->getRelativePath();
             $class = 'Symfony\\CS\\Config\\'.($relativeNamespace ? $relativeNamespace.'\\' : '').$file->getBasename('.php');
-            $this->addConfig(new $class());
+            $config = new $class();
+            if (!$config instanceof \Exception) {
```
Comment (keradus) [inline Symfony/CS/Fixer.php]: ???

Comment (SpacePossum) [inline Symfony/CS/Fixer.php]: `Symfony/CS/Config/InvalidFixerConfigurationException.php`
is now loaded by this custom kind of class loader which expects implementations of 'ConfigInterface` and not `Exception`.....

Comment (keradus) [inline Symfony/CS/Fixer.php]: ouch, that's sad

Comment (SpacePossum) [inline Symfony/CS/Fixer.php]: do you want any changes on this one?

Comment (keradus) [inline Symfony/CS/Fixer.php]: would be great, error is not a config

**Comment by SpacePossum** [comment]
reworked :)

File: Symfony/CS/InvalidFixerConfigurationException.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS;
+
+class InvalidFixerConfigurationException extends \InvalidArgumentException
```
Comment (keradus) [inline Symfony/CS/InvalidFixerConfigurationException.php]: docblock?

**Comment by keradus** [comment]
Why not 1.12?

**Comment by SpacePossum** [comment]
will redo after https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1586

**Comment by keradus** [comment]
ping @SpacePossum (1586 is on master now)

**Comment by SpacePossum** [comment]
thank you @keradus  :+1:
PR is to be expected tomorrow for the `ExtraEmptyLinesFixer`, I'll keep it small and atomic this time around :disappointed_relieved: ;)

**Comment by keradus** [comment]
ExtraEmptyLinesFixer ?

**Comment by SpacePossum** [comment]
thats the one on master having config options but doesn't have it on 1.x

**Comment by keradus** [comment]
oh, indeed

**Pull request closed without merging**